src/nnet3/nnet-example-utils.cc

```diff
@@ -192,5 +192,200 @@ void GetComputationRequest(const Nnet &nnet,
     KALDI_ERR << "No outputs in computation request.";
 }
 
+int32 NumFramesPerChunk(const NnetIo &io) {
+  std::vector<Index>::const_iterator begin = io.indexes.begin(),
+                                     iter = begin,
+                                     end = io.indexes.end();
+  unordered_set<int32> frame_indexes_t;
+  int32 n_offset = begin->n;
+  for (; iter != end; ++iter)
+    if (iter->n == n_offset)
+      frame_indexes_t.insert(iter->t);
+  return static_cast<int32>(frame_indexes_t.size());
+}
+
+int32 NumChunks(const NnetIo &io) {
+  std::vector<Index>::const_iterator begin = io.indexes.begin(),
+                                     iter = begin,
+                                     end = io.indexes.end();
+  int32 n_offset = begin->n, n = n_offset;
+  for (; iter != end; ++iter)
+    n = std::max(iter->n, n);
+  return n - n_offset + 1;
+}
+
+void SplitChunk(int32 new_chunk_size, int32 left_context, int32 right_context,
```

Comment:
Ideal size of a function is 40 lines. 
https://google.github.io/styleguide/cppguide.html#Write_Short_Functions